File: Easy/ReversePrefixofWord.java
package Easy;

/**
 * ReversePrefixofWord
 * https://leetcode.com/problems/reverse-prefix-of-word/
 */
public class ReversePrefixofWord {
    public static String reversePrefix(String word, char ch) {

        int index = word.indexOf(ch);
        if(index != -1){
            StringBuilder res1 = new StringBuilder(word.substring(0, index+1)).reverse();
            String res2 = word.substring(index+1, word.length());

            return res1.toString() + res2;

        }
        else{
            return word;
        }
    }
    public static void main(String[] args) {
        String word = "abcdefd";
        char ch = 'd';
        System.out.println(reversePrefix(word, ch));
    }
    
}

File: Easy/ReversePrefixofWord.py
# https://leetcode.com/problems/reverse-prefix-of-word/

class Solution:
    def reversePrefix(self, word: str, ch: str) -> str:
        if ch in word:
            index = word.index(ch)
            result1 = word[:index+1]
            result2 = word[index+1:]
            
            return result1[::-1] + result2
        else:
            return word
    
obj = Solution()
word = "abcdefd"
ch = "d"
print(obj.reversePrefix(word, ch))

# dcbaefd